backend/src/post/post.service.ts

```diff
@@ -33,17 +34,29 @@ export class PostService {
     }
 
 
-    async createPost(reatePostDTO:createPostDTO){
+    async createPost(reatePostDTO:createPostDTO, userData: ReqUserInfo){
+        
         const newPost = await this.prismaService.post.create({
             data:{
                 content: reatePostDTO.content,
-                userId: reatePostDTO.userId,
+                userId: userData.id,
             },
         });
         return newPost;
     }
 
-    async deletePost(postId:number){
+    async deletePost(postId:number, userId:number){
+        // データがあるかチェック
+        const post = await this.prismaService.post.findFirst({
+            where: {id: postId}
+        })
+        if (!post) {
+            throw new NotFoundException("カスタムエラー:投稿がありません");
+        }
+        if (post.userId != userId) {
+            throw new ForbiddenException("カスタムエラー:userが一致しません");
+        }
+
         const deleteUserId = await this.prismaService.post.delete({
```

Comment:
返したいものに合わせて名前の修正をお願いします！
```suggestion
        const deletePost = await this.prismaService.post.delete({
```